org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/dom/ProcessingInstruction.java

```diff
@@ -28,8 +29,22 @@ public boolean isProlog() {
 		return prolog;
 	}
 
+	public boolean isProcessingInstruction() {
+		return processingInstruction;
+	}
+
+	public void setProlog() {
+		this.prolog = true;
+		this.processingInstruction = false;
+	}
+
+	public void setProcessingInstruction() {

```

Comment:
Please keep my original code if you ar eOK with my constants PROCESSING_INSTRUCTION_NODE. More I don't want to have setProlog and setProcessingInstruction to avoid external code update it. The only compoent which must manage those flag is the parser.